src/main/kotlin/com/cosmotech/connector/adt/impl/AzureDigitalTwinsConnector.kt

```diff
@@ -80,15 +83,16 @@ class AzureDigitalTwinsConnector : Connector<DigitalTwinsClient,List<CsvData>,Li
                         dataToExport
                     )
 
-                val currentRelationships =
-                    client
-                        .listRelationships(dtInstance.id, BasicRelationship::class.java)
-                        .toList()
-                        .groupBy { it.name }
+                // Find all relationships for dtInstance.id
```

Comment:
> Do we still need to do that here? The exported data of relationships are not tied to the source twin (at least in the CSV data format we export) Now that we can query relationship independently from every specific twins we should be able to handle them completely separately from the twins themselves.

I agree. This not needed here. 
@mahkoCosmo It would be great if you could just pass the `allRelationships` collection (to group by name) to the `AzureDigitalTwinsUtil.constructRelationshipInformation` method, outside of this loop.

Reply:
> @rm3l Related to this PR (which is mostly about improving performance), is there a reason to get twins by models (using a where is_of_model query) instead of fetching all twins at once (no filter on the query) and using their metadata to dispatch the data appropriately.
> 
> Not filtering on the model type should reduce the number of API calls and improve even more the performances. Should we do that in that issue/PR while we are at it?

We can handle this in a separate issue. [PROD-7683](https://spaceport.cosmotech.com/jira/browse/PROD-7683) is more about fetching relationships. I'll create a separate issue for the point you mentioned.